agent/consul/state/catalog_test.go

```diff
@@ -3636,6 +3635,449 @@ func TestStateStore_CheckConnectServiceNodes_Gateways(t *testing.T) {
+	// 	db service on node foo with foo's node check and db's service check
+	expect := structs.CheckServiceNodes{
+		{
+			Node: &structs.Node{
+				Node:    "bar",
+				Address: "127.0.0.2",
+				RaftIndex: structs.RaftIndex{
+					CreateIndex: 11,
+					ModifyIndex: 11,
+				},
+			},
+			Service: &structs.NodeService{
+				ID:      "api",
+				Service: "api",
+				Port:    5000,
+				Weights: &structs.Weights{Passing: 1, Warning: 1},
+				RaftIndex: structs.RaftIndex{
+					CreateIndex: 20,
+					ModifyIndex: 20,
+				},
+			},
+			Checks: structs.HealthChecks{
+				{
+					Node:    "bar",
+					CheckID: "bar-node-check",
+					Status:  "passing",
+					RaftIndex: structs.RaftIndex{
+						CreateIndex: 14,
+						ModifyIndex: 14,
+					},
+				}, {
+					Node:        "bar",
+					CheckID:     "api-check",
+					Status:      "passing",
+					ServiceID:   "api",
+					ServiceName: "api",
+					RaftIndex: structs.RaftIndex{
+						CreateIndex: 21,
+						ModifyIndex: 21,
+					},
+				},
+			},
+		},
+		{
+			Node: &structs.Node{
+				Node:    "bar",
+				Address: "127.0.0.2",
+				RaftIndex: structs.RaftIndex{
+					CreateIndex: 11,
+					ModifyIndex: 11,
+				},
+			},
+			Service: &structs.NodeService{
+				ID:      "db2",
+				Service: "db",
+				Tags:    []string{"replica"},
+				Port:    8001,
+				Weights: &structs.Weights{Passing: 1, Warning: 1},
+				RaftIndex: structs.RaftIndex{
+					CreateIndex: 18,
+					ModifyIndex: 18,
+				},
+			},
+			Checks: structs.HealthChecks{
+				{
+					Node:    "bar",
+					CheckID: "bar-node-check",
+					Status:  "passing",
+					RaftIndex: structs.RaftIndex{
+						CreateIndex: 14,
+						ModifyIndex: 14,
+					},
+				},
+				{
+					Node:        "bar",
+					CheckID:     "db2-check",
+					Status:      "passing",
+					ServiceID:   "db2",
+					ServiceName: "db",
+					ServiceTags: []string{"replica"},
+					RaftIndex: structs.RaftIndex{
+						CreateIndex: 19,
+						ModifyIndex: 19,
+					},
+				},
+			},
+		},
+		{
+			Node: &structs.Node{
+				Node:    "foo",
+				Address: "127.0.0.1",
+				RaftIndex: structs.RaftIndex{
+					CreateIndex: 10,
+					ModifyIndex: 10,
+				},
+			},
+			Service: &structs.NodeService{
+				ID:      "db",
+				Service: "db",
+				Port:    5000,
+				Weights: &structs.Weights{Passing: 1, Warning: 1},
+				RaftIndex: structs.RaftIndex{
+					CreateIndex: 12,
+					ModifyIndex: 12,
+				},
+			},
+			Checks: structs.HealthChecks{
+				{
+					Node:    "foo",
+					CheckID: "foo-node-check",
+					Status:  "passing",
+					RaftIndex: structs.RaftIndex{
+						CreateIndex: 13,
+						ModifyIndex: 13,
+					},
+				},
+				{
+					Node:        "foo",
+					CheckID:     "db-check",
+					Status:      "passing",
+					ServiceID:   "db",
+					ServiceName: "db",
+					RaftIndex: structs.RaftIndex{
+						CreateIndex: 15,
+						ModifyIndex: 15,
+					},
+				},
+			},
+		},
+	}
+	assert.Equal(expect, nodes)
```

Comment:
```suggestion
	assert.ElementsMatch(expect, nodes)
```

This is a bit more flexible as it will not require any specific ordering.